lib/screens/auth/google_oauth.dart

```diff
@@ -0,0 +1,162 @@
+                  ElevatedButton(
+                    onPressed: _handleGoogleSignOut,
+                    style: ElevatedButton.styleFrom(
+                      foregroundColor: Colors.white,
+                      backgroundColor: Colors.red,
+                      minimumSize: const Size(100, 50),
+                    ),
+                    child: const Text('Logout'),
+                  ),
+                ],
+              ),
+            )
+          : Center(
+              child: MaterialButton(
+                onPressed: _handleGoogleSignIn,
+                height: 50,
+                minWidth: 200,
+                color: Colors.blue,
+                textColor: Colors.white,
+                child: const Text('Sign in with Google'),
+              ),
+            ),
+    );
+  }
+
+  Future<void> _handleGoogleSignIn() async {
+    try {
+      final GoogleSignInAccount? googleUser = await _googleSignIn.signIn();
+      if (googleUser != null) {
+        setState(() {
+          _isLoggedIn = true;
+          _userObj = googleUser;
+        });
+
+        final QuerySnapshot querySnapshot = await _firestore
+            .collection('users')
+            .where('userId', isEqualTo: _userObj.id)
+            .get();
+        if (querySnapshot.docs.isNotEmpty) {
+          final userData = querySnapshot.docs.first.data();
+          final isFirstLogin =
+              (userData as Map<String, dynamic>)['is_first_login'] ?? true;
+
+          if (isFirstLogin) {
+            Navigator.pushReplacement(
+              context,
+              MaterialPageRoute(
+                builder: (context) => DataEntryPage(
+                  email: _userObj.email,
+                ),
+              ),
+            );
+          } else {
+            sessionManager.setUserId(_userObj.id);
+            Navigator.pushReplacement(
+              context,
+              MaterialPageRoute(
+                builder: (context) => UserHomePage(userData: {
+                  'name': _userObj.displayName,
+                  'email': _userObj.email,
+                  'user_id': _userObj.id,
+                }),
+              ),
+            );
+          }
+        } else {
+          await _saveUserDataToFirestore(_userObj);
+
+          Navigator.pushReplacement(
+            context,
+            MaterialPageRoute(
+              builder: (context) => DataEntryPage(
+                email: _userObj.email,
+              ),
+            ),
+          );
+        }
+      }
+    } catch (error) {
+      print('Error signing in with Google: $error');
+    }
+  }
+
+  Future<void> _handleGoogleSignOut() async {
+    try {
+      await _googleSignIn.signOut();
+
+      await _authService.signOut();
+
+      final prefs = await SharedPreferences.getInstance();
+      await prefs.clear();
+
+      setState(() {
+        _isLoggedIn = false;
+      });
+    } catch (error) {
+      print('Error signing out: $error');
+    }
+  }
```

Comment:
If the user fails to register for any reason, will they not receive any notification?